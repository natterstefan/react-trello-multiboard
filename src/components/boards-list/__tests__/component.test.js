import React from 'react'
import { shallow } from 'enzyme'
import LinearProgress from '@material-ui/core/LinearProgress'

import BoardsList from '../component'
import Board from '../../board'

const props = {
  error: null,
  boards: [
    {
      board: {
        name: 'board-1',
      },
      config: {},
    },
  ],
  getEstimations: () => ({
    estimated: 3,
    consumed: 2,
  }),
  isLoading: false,
}

describe('Component/BoardsList', () => {
  test('should render without throwing an error', () => {
    expect(shallow(<BoardsList {...props} />)).toMatchSnapshot()
  })

  test('should render an LinearProgress when component is waiting for loading data to finish', () => {
    const wrapper = shallow(<BoardsList {...props} isLoading />)
    expect(wrapper.find(LinearProgress).length).toEqual(1)
  })

  test('should render an empty span when an error occured', () => {
    const wrapper = shallow(<BoardsList {...props} error="sample error" />)
    expect(wrapper.html()).toEqual('<span></span>')
  })

  test('should render a Board', () => {
    const wrapper = shallow(<BoardsList {...props} />)
    expect(wrapper.find(Board).length).toEqual(1)
  })

  test('should pass the correct properties to the rendered Board', () => {
    const wrapper = shallow(<BoardsList {...props} />)
    expect(wrapper.find(Board).props()).toMatchObject({ board: { name: 'board-1' }, config: {} })
  })
})
