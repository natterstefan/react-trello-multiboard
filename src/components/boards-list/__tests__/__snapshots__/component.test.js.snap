// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component/BoardsList should render without throwing an error 1`] = `
<styled.div>
  <styled.div
    key="board-1"
  >
    <WithStyles(Typography)
      variant="headline"
    >
      board-1
       
      <span>
        (
        3
        )[
        2
        ]
      </span>
    </WithStyles(Typography)>
    <Connect(Board)
      board={
        Object {
          "id": "board-1",
          "name": "board-1",
        }
      }
      config={Object {}}
    />
  </styled.div>
</styled.div>
`;
